source/guides/basics/loops.md

```diff
@@ -46,7 +139,74 @@ my_move_task:
         - wait 2s
 ```
 
-You can try this in-game via `/ex run my_move_task`. It will repeatedly ask you to move until you move at least 3 blocks away.
+You can try this in-game via `/ex run my_move_task`. This task when run will keep telling you to move away until you have moved 3 blocks away from when you first were.
+
+Just like the `repeat` command we have a `wait` command at the end of the loop. While it can be optional for `repeat` loops to use this, it is almost always required for `while` loops. Without it it will try to check as much as it can all at the same time, which can cause the server to crash.
+
+As a word of warning, `while` loops should be avoided if possible. It is very easy to make what is known as an infinite loop, as in the loop will never have a chance to stop. Once its running, it is very hard to force it to stop beyond just stopping the server.
```

Comment:
```suggestion
Just like the `repeat` command, we have a `wait` command at the end of the loop. While it's only useful in some cases with `repeat` loops, you should almost always use it with `while` loops. Without one, it will try to check the condition as often as possible all at the same time, which can cause the server to crash.

A word of warning: `while` loops should be avoided if possible. It is very easy to make what is known as an "infinite loop", as in a loop that never has a chance to stop. Once it's running, it's difficult to force it to stop beyond just stopping the server.
```